tests/foreman/cli/test_syncplan.py

```diff
@@ -155,24 +154,25 @@ def validate_repo_content(self, repo, content_types, after_sync=True):
         repo = Repository.info({'id': repo['id']})
         for content in content_types:
             if after_sync:
-                self.assertGreater(int(repo['content-counts'][content]), 0)
+                assert int(repo['content-counts'][content]) > 0
             else:
-                self.assertFalse(int(repo['content-counts'][content]))
+                assert not int(repo['content-counts'][content])
 
+    @pytest.mark.parametrize('name', **parametrized(valid_data_list()))
     @tier1
-    def test_positive_create_with_name(self):
+    def test_positive_create_with_name(self, name):
         """Check if syncplan can be created with random names
 
         :id: dc0a86f7-4219-427e-92fd-29352dbdbfce
 
+        :parametrized: yes
+
         :expectedresults: Sync plan is created and has random name
 
         :CaseImportance: Critical
         """
-        for name in valid_data_list():
-            with self.subTest(name):
-                new_sync_plan = self._make_sync_plan({'name': name})
-                self.assertEqual(new_sync_plan['name'], name)
+        sync_plan = self._make_sync_plan({'name': name})
+        assert sync_plan['name'] == name
```

Comment:
Is this actually asserting against the name that was set on the entity within Satellite, or is it asserting against the _object_ returned by this function?  Its an important distinction here, and the test should be fetching the entity after it is created to strongly assert that the attribute was correct on Satellite itself.

Reply:
Will check

Reply:
@mshriver I fixed the first three test by adding `result = SyncPlan.info` to read back the Sync plan before making an assert.

Reply:
Great! That's a much stronger assertion pattern for this test. We don't want to just be testing our entity class constructor :)